src/middleware/session/mod.rs

```diff
@@ -51,49 +51,73 @@ enum SessionDataState {
     Dirty,
 }
 
-#[derive(Copy, Clone, PartialEq)]
+#[derive(Copy, Clone, PartialEq, Debug)]
 enum SameSiteEnforcement {
     Strict,
     Lax,
 }
 
-#[derive(Clone, PartialEq)]
-enum CookieOption {
-    // If `Expires` / `Max-Age` are ever added here, be sure to update `reset_session` to allow
-    // for them.
-    Secure,
-    HttpOnly,
-    SameSite(SameSiteEnforcement),
-    Domain(String),
-}
-
-impl fmt::Display for CookieOption {
-    fn fmt(&self, out: &mut fmt::Formatter) -> fmt::Result {
-        use self::CookieOption::*;
-        use self::SameSiteEnforcement::*;
-
-        match *self {
-            Secure => out.write_str("secure"),
-            HttpOnly => out.write_str("HttpOnly"),
-            SameSite(Strict) => out.write_str("SameSite=strict"),
-            SameSite(Lax) => out.write_str("SameSite=lax"),
-            Domain(ref s) => {
-                out.write_str("Domain=")?;
-                out.write_str(s)
-            }
-        }
-    }
-}
-
 /// Configuration for how the `Set-Cookie` header is generated.
 ///
 /// By default, the cookie has the name "_gotham_session", and the cookie header includes the
 /// `secure` flag.  `NewSessionMiddleware` provides functions for adjusting the
 /// `SessionCookieConfig`.
 #[derive(Clone)]
 pub struct SessionCookieConfig {
+    // If `Expires` / `Max-Age` are ever added update `reset_session` to allow for them.
     name: String,
-    options: Vec<CookieOption>,
+    secure: bool,
+    http_only: bool,
+    same_site: SameSiteEnforcement,
+    path: String,
+    domain: Option<String>,
+}
+
+impl Default for SessionCookieConfig {
+    fn default() -> SessionCookieConfig {
+        SessionCookieConfig {
+            name: "_gotham_session".to_string(),
+            secure: true,
+            http_only: true,
+            same_site: SameSiteEnforcement::Strict,
```

Comment:
The prior default was `SameSiteEnforcement::Lax`, which was chosen after careful deliberation. I propose changing it back unless there's a compelling reason for the new default.

Seems likely that this is an oversight, because the `with_strict_same_site_enforcement` function still exists but is a noop now.